ls1intum/Artemis
# [ Enhancement ] - Add functionalities to edit selected

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest2.ase.in.tum.de and https://vmbruegge60.in.tum.de/.

### Motivation and Context


Some feedbacks from instructors using Artemis asked about the possibility of adding the **Export Results** functionality to the already implemented **edit selected** functionality. 

### Description

* The current **Export Results** functionality in `exercise-scores.component.ts` has been extracted into an individual component `exercise-scores-export-button.component.ts` which is then reused in the *Scores view* and *Programming exercise list view*.

* To better distinguish the downloaded results files, the *shortName* of the exercise has been appended to the file name

* In addition, the select/deselect all, functionality has been added.

* Added warning if exercise **without** results is exported

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Exercises of a Course
4. Try the select/deselect button (see screenshot)
5. Select some exercises and export the results
6. Navigate to the score view of an exercise
7. Check if scores are still displayed

### Test Coverage



exercise-scores-export-button.component.ts 87%

### Screenshots


![Screenshot from 2021-04-21 16-46-41](https://user-images.githubusercontent.com/45761921/115573731-4ed67f00-a2c1-11eb-9122-84d6f28a5ccd.png)
![Screenshot from 2021-04-21 16-48-15](https://user-images.githubusercontent.com/45761921/115573972-82b1a480-a2c1-11eb-8c6c-30256e9d4289.png)
